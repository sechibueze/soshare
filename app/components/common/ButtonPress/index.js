import { Pressable, Text, StyleSheet } from 'react-native';
import React from 'react';
import { STYLES } from 'config/styles.config';

export default function ButtonPress({ label, style, ...rest }) {
  return (
    <Pressable style={[styles.button, style]} {...rest}>
      {!!label && <Text style={styles.buttonText}>{label} </Text>}
    </Pressable>
  );
}
const styles = StyleSheet.create({
  button: {
    width: '85%',
    backgroundColor: STYLES.color.primary,
    padding: 12,
    borderRadius: 4,
  },
  buttonText: {
    color: STYLES.color.light,
    fontFamily: STYLES.font.font__medium,
    // fontWeight: 'bold',
    textAlign: 'center',
    fontSize: 18,
  },
});
