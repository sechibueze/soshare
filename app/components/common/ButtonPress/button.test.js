import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPress from '.';
describe('It renders', () => {
  it('It renders unit', () => {
    const tree = renderer.create(<ButtonPress title='hello' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
