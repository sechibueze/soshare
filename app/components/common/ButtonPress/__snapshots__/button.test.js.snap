// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders It renders 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "blue",
        "borderRadius": 4,
        "padding": 12,
        "width": "85%",
      },
      undefined,
    ]
  }
  title="hello"
/>
`;
